### PR TITLE
3point7

### PR DESCRIPTION
**Pull request by DavidJFigueroa**

### CODE REVIEW

File: src/components/profile-view/profile-view.jsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+          <FavoriteMovies
+            movies={movies}
+            user={user}
+            removeFavorite={removeFavorite}
+          />
+        </Col>
+      </Row>
+    </Container>
+  );
+};
```
Comment (stratospherique) [inline src/components/profile-view/profile-view.jsx:117]: Missing closing bracket `}`

File: src/components/profile-view/profile-view.jsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+  return (
+    <Container>
+      <Row>
+        <Col xs={12} sm={6}>
+          {" "}
+          <UserInfo
+            name={user.Username}
+            email={user.Email}
+            handleSubmitDeregister={handleSubmitDeregister}
```
Comment (stratospherique) [inline src/components/profile-view/profile-view.jsx]: 1. If you check your API, you'll notice that the endpoint for removing a user doesn't require a payload in the request body.
2. The function `handleSubmitDeregister` was defined with two parameters but was called without any argument(implicitly with the event argument). Regardless, that function should only call the remove user endpoint then call the `onLoggout` function defined in the main-view(you can pass it as a prop). Then you can safely redirect to the signup path.

File: src/components/profile-view/update-user.jsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+          </Row>
+
+          <Button variant="primary" type="submit">
+            Update
+          </Button>
+        </Form>
+      </Card.Body>
+    </Card>
+  );
+}
```
Comment (stratospherique) [inline src/components/profile-view/update-user.jsx:86]: proptypes definition missing

Comment (stratospherique) [inline src/components/profile-view/update-user.jsx:86]: proptypes definition missing

File: src/components/profile-view/user-info.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+          </p>
+
+          <Button variant="primary" onClick={handleSubmitDeregister}>
+            Delete Account
+          </Button>
+        </Card.Body>
+      </Card>
+    </>
+  );
+}
```
Comment (stratospherique) [inline src/components/profile-view/user-info.jsx:26]: proptypes definition missing

File: src/components/profile-view/profile-view.jsx (hunk 1 above)
Comment (stratospherique) [reply src/components/profile-view/profile-view.jsx:117]: proptypes definition missing

**Review by stratospherique — CHANGES_REQUESTED** [review]
Still needs revision.

File: src/components/profile-view/profile-view.jsx (hunk 2 above)
Comment (DavidJFigueroa) [reply src/components/profile-view/profile-view.jsx]: 1. so i just remove: body: JSON.stringify(data)?
2. do i call the function here:
`    <UserInfo
            name={user.Username}
            email={user.Email}
            handleSubmitDeregister={handleSubmitDeregister(and which argument do i need to put here?);}`
4. how do i call the remove user endpoint? 
5.  i already passed the onLoggedOut function as prop, does it need to be called with an argument aswell?

Comment (DavidJFigueroa) [reply src/components/profile-view/profile-view.jsx]: Ok, so i changed it to now:
`
    fetch(
      `https://myflix-database-api-9ba401fe0e70.herokuapp.com/users/${user.Username}`,
      {
        method: "DELETE",
        headers: {
          "Content-Type": "application/json",
          Authorization: `Bearer ${token}`,
        },
      }
    ).then((response) => {
      if (response.ok) {
        onLoggedOut();
        window.location.replace("/signup");
      } else {
        alert("Deregister failed");
      }
    });
  };`

and i pass the onLoggedOut={handleLogout} as props from main-view to profile-view. but it gives me onLoggedOut is not a function. Thats why i need to give it an argument right? 

I defined the function in the main-view like this
`const handleLogout = () => {
    setUser(null);
    setToken(null);
    localStorage.clear();
  };`

Which arguments do i need to call it with?

File: src/components/profile-view/favorite-movies.jsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+              removeFavorite={() => removeFavorite(movie)}
+            />
+          </Col>
+        ))}
+      </Row>
+    </>
+  );
+}
+
+FavoriteMovies.PropTypes = {
```
Comment (stratospherique) [inline src/components/profile-view/favorite-movies.jsx]: This line and many similar ones in other components are causing the app to crash. It's `.proptypes` with a lowercase p.

**Review by stratospherique — CHANGES_REQUESTED** [review]

**Pull request merged**